client/src/Counter.js

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react"
+const Counter = ({ setAllMyVideos, videoId, videoRating }) => {
+    const [rating, setRating] = useState(videoRating)
+    let newRating
+    const handleCounterDown = () => {
+        if (rating <= 0) {
+            newRating = 0
+            setRating(newRating)
+            handleRatingChange(newRating)
+            getAllNewVideos()
+        } else {
+            newRating = rating - 1
+            setRating(newRating)
+            handleRatingChange(newRating)
+            getAllNewVideos()
+        }
```

Comment:
Your two branches here mostly do the same thing - I would factor out the common things outside of the branch:
```suggestion
        if (rating <= 0) {
            newRating = 0
        } else {
            newRating = rating - 1
        }
        setRating(newRating)
        handleRatingChange(newRating)
        getAllNewVideos()
```

There are two reasons for this:
1. If we need to change what we do, we will only need to change it in one place (and can't forget to change the other)
2. It makes it more clear to the reader of the code what's _actually_ different, and what's the same, in the two cases.